dll/win32/shdocvw/CNSCBand.cpp

```diff
@@ -1246,14 +1297,45 @@ STDMETHODIMP CNSCBand::HasFocusIO()
 
 STDMETHODIMP CNSCBand::TranslateAcceleratorIO(LPMSG lpMsg)
 {
+    if (lpMsg->message == WM_KEYDOWN && lpMsg->wParam == VK_F2 && !m_isEditing)
+    {
+        if (HTREEITEM hItem = m_hwndTreeView.GetNextItem(NULL, TVGN_CARET))
+        {
+            if (_GetAttributesOfItem(_GetItemData(hItem), SFGAO_CANRENAME))
+            {
+                m_hwndTreeView.SetFocus();
+                m_hwndTreeView.EditLabel(hItem);
+                return S_OK;
+            }
+        }
+    }
+
+    if (lpMsg->message == WM_SYSKEYDOWN && lpMsg->wParam == VK_RETURN && !m_isEditing)
+    {
+        CItemData *pItem = _GetItemData(TVGN_CARET);
+        if (pItem && _GetAttributesOfItem(pItem, SFGAO_HASPROPSHEET))
+        {
+            SHELLEXECUTEINFOW sei = { sizeof(sei), SEE_MASK_INVOKEIDLIST, m_hwndTreeView };
+            sei.lpVerb = L"properties";
+            sei.lpIDList = pItem->absolutePidl;
+            sei.nShow = SW_SHOW;
+            ShellExecuteExW(&sei);
+            return S_OK;
+        }
+        else
+        {
+            MessageBeep(0);
```

Comment:
```suggestion
            MessageBeep(MB_OK);
```
Should this be an "ok" sound, or a warning or error one instead?